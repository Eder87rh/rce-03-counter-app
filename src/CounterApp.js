import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  const handleApp = () => {
    setCounter(counter + 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  const handleDecrement = () => {
    setCounter(counter - 1);
  };

  return (
    <div>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleApp}>+1</button>
      <button onClick={handleReset}>reset</button>
      <button onClick={handleDecrement}>-1</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
