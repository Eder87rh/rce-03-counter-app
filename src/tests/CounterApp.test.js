import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("CounterApp", () => {
  it("Counter app should display correctly", () => {
    const wrapper = shallow(<CounterApp value={10} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("Counter app should display 10 by default", () => {
    const wrapper = shallow(<CounterApp />);
    const value = wrapper.find("h2").text();
    expect(value).toBe("10");
  });
});
