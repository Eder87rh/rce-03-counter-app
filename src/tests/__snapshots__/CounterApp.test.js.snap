// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Counter app should display correctly 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
